extra_data/keydata.py

```diff
@@ -396,6 +416,14 @@ def ndarray(self, roi=(), out=None):
             )
             dest_cursor = dest_chunk_end
 
+        if out.dtype.hasobject:
+            # Can current only occur for string properties, convert from
+            # object array of bytes to to object array of strings.
+            out = np.array(
+                [x.decode('utf8', 'surrogateescape') for x in out.flat],
+                dtype=object
+            ).reshape(out.shape)
```

Comment:
This will fail if we ever have a compound dtype containing a string field. But that seems unlikely and it's not trivial to do the right thing in that case, so let's leave it as is and cross that bridge if we come to it.

Reply:
Hu, well spotted. Better safe than sorry, I will check `dtype.fields` for this branch as well. In this weird case we then simply opt out of string treatment.